Automattic/wp-calypso
# Subscriptions: Grouping Newsletter settings

## Proposed Changes

It groups the Newsletter Subscriptions settings to be consistent with what we have in Jetpack.

### Before

<img width="740" alt="Screenshot 2024-06-03 at 17 00 02" src="https://github.com/Automattic/wp-calypso/assets/4068554/ab17349b-6ff5-443f-ab1a-432779b7f98c">

### After

<img width="740" alt="Screenshot 2024-06-03 at 16 58 16" src="https://github.com/Automattic/wp-calypso/assets/4068554/a37c85ec-7ad6-4429-85b6-df8b6202e63d">

## Testing Instructions

Go to the Newsletter settings and make sure the Subscriptions settings work as expected.

## Pre-merge Checklist
